lib/rubocop/cop/rspec/focus.rb

```diff
@@ -22,19 +22,22 @@ module RSpec
       class Focus < Base
         MSG = 'Focused spec found.'
 
-        focused = ExampleGroups::FOCUSED + Examples::FOCUSED
-
-        def_node_matcher :focusable_selector?,
-                         (ExampleGroups::GROUPS + ExampleGroups::SKIPPED +
-                          Examples::EXAMPLES + Examples::SKIPPED +
-                          Examples::PENDING).node_pattern_union
+        def_node_matcher :focusable_selector?, <<-PATTERN
+          {#rspec(:ExampleGroups, :Regular) #rspec(:ExampleGroups, :Skipped)
```

Comment:
@marcandre Do you think it makes sense (for other core cops and extension cops) to provide this kind of syntax:
```ruby
def_node_matcher :focusable_selector?, <<-PATTERN
  #rspec.example_groups.regular
PATTERN
```
e.g. so that the Node Pattern's function calls/predicates can be namespaced/chained?

Disclaimer: there's only an aestetic effect, no actual breakthroughs of having such a syntax. By "makes sense" I mean "is it worth the effort of implementing/maintaining this additional complexity".

Reply:
It could be interesting to have namespaces somehow; I'd be enclined to use constants though... `RSpec::ExampleGroups.regular`, assuming `regular` is a singleton_method; and maybe `def_node_matcher` should define a singleton_class method at the same time.